site/src/Librecores/ProjectRepoBundle/RepoCrawler/GitRepoCrawler.php

```diff
@@ -88,6 +92,17 @@ class GitRepoCrawler extends RepoCrawler
 
     private $repoClonePath = null;
 
+    public function __construct(
+        SourceRepo $repo,
+        MarkupToHtmlConverter $markupConverter,
+        ProcessCreator $processCreator,
+        ObjectManager $manager,
+        LoggerInterface $logger,
+        $projectMetricsProvider
```

Comment:
Is there a reason for this parameter not having a type annotation?

Reply:
Both of the approaches will work because projectmetrics are initialized from the RepoCrawler itself thru the constructor. So, I removed the effort of requiring the ProjectMetricsProvider class always and injecting it.